### PR TITLE
User/amohdsharfuddin/dkip 24

### PR DESCRIPTION
**Pull request by amohdsharfuddin**
- Added a landing page with About Us view and a couple of browse by category carousel views
- Added a search on the default layout with a search function
- The view re-renders when the search input is focused/blurred

### CODE REVIEW

**Comment by cthye-gw** [comment]
Hi @amohdsharfuddin , I saw some files' changes are just the spacing and removing unused imports. Can I confirm with you that you've run some code analysis or automation to perform all these changes??

**Review by amohdsharfuddin — COMMENTED** [review]
Fixes:

- Updated React to latest version
- Updated landing page view to implement search function
- Fixed package dependencies issues
- Installed missing packages

**Comment by amohdsharfuddin** [comment]
> Hi @amohdsharfuddin , I saw some files' changes are just the spacing and removing unused imports. Can I confirm with you that you've run some code analysis or automation to perform all these changes??

HI @cthye-gw , I apologize for my mistake. I did not test the code out before pushing. So, I have committed a new fix after I did run the codes and take note all of the running issues. I did have an extension that auto formats my file when I save it so maybe that is whats making the spaces and import changes. Should I discard the extension?

**Comment by cthye-gw** [comment]
> > Hi @amohdsharfuddin , I saw some files' changes are just the spacing and removing unused imports. Can I confirm with you that you've run some code analysis or automation to perform all these changes??
> 
> HI @cthye-gw , I apologize for my mistake. I did not test the code out before pushing. So, I have committed a new fix after I did run the codes and take note all of the running issues. I did have an extension that auto formats my file when I save it so maybe that is whats making the spaces and import changes. Should I discard the extension?

Yes, that should be it. make sure you let @jliow know about it and document in the guide & Readme (if exist) so that spacing formats follow the same standards.

**Review by kchang-gw — COMMENTED** [review]
1. When click on search function, it open up a drawer. Is it design as expected?
2. Please fix the warning message highlighted in red in console.

File: src/components/SignList.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+const SignList = ({ filter }) => {
+  const classes = classNames("mt-3");
+
+  return (
+    <div className={classes}>
+      {Store.getSignItems()
+        // eslint-disable-next-line array-callback-return
+        .filter((val) => {
+          if (filter === "") return val;
```
Comment (kchang-gw) [inline src/components/SignList.js]: Let's add the open close braces even it is only one line of code to execute after it. This is to make the code easy to read and maintain.

**Comment by amohdsharfuddin** [comment]
> 1. When click on search function, it open up a drawer. Is it design as expected?
> 2. Please fix the warning message highlighted in red in console.

1. Its not technically a drawer. The whole page re-renders which is triggered by the search input's focus event.
2. Will do ASAP with a new commit.

File: src/components/SignList.js (hunk 1 above)
Comment (amohdsharfuddin) [reply src/components/SignList.js]: Alright. I will add that in an upcoming commit.

**Review by amohdsharfuddin — COMMENTED** [review]
Bug fixes:

- Fix console warning ( unique key on map array function)
- Added open close braces in SignList.js at line 20 and 25
- Update search input field placeholder
- Added updated icons with logo 
- Update package dependencies

**Review by amohdsharfuddin — COMMENTED** [review]
Added a Homepage as a landing page

**Review by kchang-gw — APPROVED** [review]

**Pull request merged**